migrations/V9__Plan_Collaborator_Lead_Trigger.sql

```diff
@@ -0,0 +1,29 @@
+CREATE FUNCTION collaborator_role_check_trigger() RETURNS TRIGGER AS $role_check$
+BEGIN
+ IF (SELECT count(*)
+ FROM plan_collaborator
+ WHERE  team_role = 'MODEL_LEAD' and model_plan_id = OLD.model_plan_id 
+ )<2
+ THEN
+    RAISE EXCEPTION 'There must be at least one MODEL_LEAD assigned to each model plan';
+     END IF;
+     IF (TG_OP = 'DELETE') THEN RETURN OLD;
+     ELSE
+    RETURN NEW;
+    END IF;
+END
+$role_check$ LANGUAGE plpgsql;
+
+
+CREATE TRIGGER collaborator_lead_req_update
+    BEFORE UPDATE ON plan_collaborator
+	FOR EACH ROW
+    WHEN ((OLD.team_role = 'MODEL_LEAD') AND (NEW.team_role != 'MODEL_LEAD'))
```

Comment:
My knowledge of SQL triggers is pretty sparse - should this be an `OR` instead of an `AND`?

Reply:
I don't think so for this situation, I only want to trigger this statement if the row that I'm updating was already a MODEL_LEAD and is changing to something that isn't a model lead.  Or will trigger this statement a lot more than is needed in this situation.

Reply:
Derp - I misread `!=` as `=`.